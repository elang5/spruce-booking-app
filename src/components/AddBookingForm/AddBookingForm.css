.booking-form {
  display: flex;
  justify-content: center;
}

.input-cell > input, .input-cell > select {
  padding: 7px;
  margin: 5px 0;
  border-radius: 2px;
  border: solid 1px #979797;
  margin-bottom: 10px;
  width: 265px;
  box-sizing: border-box;
  font-size: 14px;
}

.input-cell > input {
  padding-left: 5px;
}

.input-cell > select {
  height: 33px;
  width: 100%;
}

.input-cell > label {
  font-size: 12px;
  letter-spacing: 0.96px;
  margin-left: 2px;
}

.col-1 {
  margin-right: 70px;
}

.input-cell, .col-1, .col-2 {
  display: flex;
  flex-direction: column;
}

.booking-form > button {
  position: absolute;
  bottom: 20px;
  right: 5px;
}

.state-zip-container {
  display: flex;
  width: 100%;
  justify-content: space-between;
}

.state-zip-container > .input-cell > input {
  max-width: 120px;
}
