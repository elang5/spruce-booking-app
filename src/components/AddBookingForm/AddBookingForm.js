import React from "react"
import Loading from "../Loading/Loading"
import "./AddBookingForm.css"

export default function AddBookingForm({ handleSubmit, handleFields, fields, isLoading }) {
  return (
    <form className="booking-form" onSubmit={e => handleSubmit(e)}>
      <div className="col-1">
        <div className="input-cell">
          <label htmlFor="name">Name</label>
          <input id="name" type="text" value={fields.name || ""} name="name" onChange={handleFields}/>
        </div>
        <div className="input-cell">
          <label htmlFor="email">Email</label>
          <input id="email" type="text" value={fields.email || ""} name="email" onChange={handleFields}/>
        </div>
        <div className="input-cell">
          <label htmlFor="address">Street Address</label>
          <input id="address" type="text" value={fields.address || ""} name="address" onChange={handleFields}/>
        </div>
        <div className="input-cell">
          <label htmlFor="city">City</label>
          <input id="city" type="text" value={fields.city || ""} name="city" onChange={handleFields}/>
        </div>
        <div className="state-zip-container">
          <div className="input-cell">
            <label htmlFor="state">State</label>
            <input id="state" type="text" value={fields.state || ""} name="state" onChange={handleFields}/>
          </div>
          <div className="input-cell">
            <label htmlFor="zip">Zip Code</label>
            <input id="zip" type="number" value={fields.zip || ""} name="zip" onChange={handleFields}/>
          </div>
        </div>
      </div>
      <div className="col-2">
        <div className="input-cell">
          <label htmlFor="booking-type">Booking type</label>
          <select required id="booking-type" value={fields.bookingtype || ""} name="bookingtype" onChange={handleFields}>
            <option value="housekeeping">Housekeeping</option>
            <option value="dogwalk">Dog Walk</option>
          </select>
        </div>
        <div className="input-cell">
          <label htmlFor="booking-date">Booking Date</label>
          <input required type="date" name="bookingdate" id="booking-date" value={fields.bookingdate || ""} onChange={handleFields}/>
        </div>
        <div className="input-cell">
          <label htmlFor="booking-time">Booking Time</label>
          <input required type="time" name="bookingtime" id="booking-time" value={fields.bookingtime || ""} onChange={handleFields}/>
        </div>
      <Loading loading={isLoading} />
      </div>
      <button type="submit" className="create-booking">
        Create booking
      </button>
    </form>
  )
}