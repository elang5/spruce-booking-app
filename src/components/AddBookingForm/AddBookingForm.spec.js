import React from 'react';
import ReactDOM from 'react-dom';
import AddBookingForm from './AddBookingForm';
import renderer from 'react-test-renderer';

describe('AddBookingForm Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(< AddBookingForm />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<AddBookingForm/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
})
