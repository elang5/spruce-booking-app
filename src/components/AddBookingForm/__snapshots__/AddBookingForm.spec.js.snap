// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddBookingForm Component renders the UI as expected 1`] = `
<form
  className="booking-form"
  onSubmit={[Function]}
>
  <div
    className="col-1"
  >
    <div
      className="input-cell"
    >
      <label
        htmlFor="name"
      >
        Name
      </label>
      <input
        id="name"
        name="name"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="input-cell"
    >
      <label
        htmlFor="email"
      >
        Email
      </label>
      <input
        id="email"
        name="email"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="input-cell"
    >
      <label
        htmlFor="address"
      >
        Address
      </label>
      <input
        id="address"
        name="address"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="input-cell"
    >
      <label
        htmlFor="city"
      >
        City
      </label>
      <input
        id="city"
        name="city"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="state-zip-container"
    >
      <div
        className="input-cell"
      >
        <label
          htmlFor="state"
        >
          State
        </label>
        <input
          id="state"
          name="state"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="input-cell"
      >
        <label
          htmlFor="zip"
        >
          Zip Code
        </label>
        <input
          id="zip"
          name="zip"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
  <div
    className="col-2"
  >
    <div
      className="input-cell"
    >
      <label
        htmlFor="booking-type"
      >
        Booking type
      </label>
      <select
        id="booking-type"
        name="bookingtype"
        required={true}
        value=""
      >
        <option
          value="housekeeping"
        >
          Housekeeping
        </option>
        <option
          value="dogwalk"
        >
          Dog Walk
        </option>
      </select>
    </div>
    <div
      className="input-cell"
    >
      <label
        htmlFor="booking-date"
      >
        Booking Date
      </label>
      <input
        id="booking-date"
        name="bookingdate"
        onChange={[Function]}
        required={true}
        type="date"
        value=""
      />
    </div>
    <div
      className="input-cell"
    >
      <label
        htmlFor="booking-time"
      >
        Booking Time
      </label>
      <input
        id="booking-time"
        name="bookingtime"
        onChange={[Function]}
        required={true}
        type="time"
        value=""
      />
    </div>
    <div
      className="loading-container"
    />
  </div>
  <button
    className="create-booking"
    type="submit"
  >
    Create booking
  </button>
</form>
`;
